### ISSUE TITLE
palendrome: unit test expects incorrect output

### ISSUE THREAD

**Issue by synthet1c** [FreeCodeCamp/FreeCodeCamp#9213]
Challenge [Check for Palindromes](https://www.freecodecamp.com/challenges/check-for-palindromes#?solution=function%20onlyLowerCaseChars(%20str%20)%7B%0A%20%20return%20str%0A%20%20%20%20.replace(%2F(%5B%5Ea-z%5D%2B)%2Fgmi%2C%20'')%0A%20%20%20%20.toLowerCase()%3B%0A%7D%0A%0Afunction%20palindrome(str)%20%7B%0A%20%20%2F%2F%20Good%20luck!%0A%20%20%0A%20%20if(%20!str.length%20)%7B%0A%20%20%20%20return%20false%3B%0A%20%20%7D%0A%20%20%0A%20%20var%20replaced%20%3D%20onlyLowerCaseChars(%20str%20)%3B%0A%20%20%0A%20%20var%20left%20%3D%20replaced.split('')%3B%0A%20%20var%20right%20%3D%20replaced.split('').reverse()%3B%0A%20%20%0A%20%20for(%20var%20ii%20%3D%200%3B%20ii%20%3C%20left.length%3B%20ii%2B%2B)%7B%0A%20%20%20%20if(%20left%5Bii%5D%20!%3D%3D%20right%5Bii%5D%20)%7B%0A%20%20%20%20%20%20return%20false%3B%0A%20%20%20%20%7D%0A%20%20%7D%0A%20%20%0A%20%20return%20true%3B%0A%7D%0A%0A%0A%0Apalindrome(%22eye%22)%3B%0A) has an issue.
User Agent is: <code>Mozilla/5.0 (Windows NT 6.3; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.84 Safari/537.36</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

The 10th test `palindrome("1 eye for of 1 eye.")` expects `false` but should return `true`;

result of striping out all non chars is `"eyeforofeye"` which is a palendrome.

My code:
```javascript
function onlyLowerCaseChars( str ){
  return str
    .replace(/([^a-z]+)/gmi, '')
    .toLowerCase();
}

function palindrome(str) {
  // Good luck!
  
  if( !str.length ){
    return false;
  }
  
  var replaced = onlyLowerCaseChars( str );
  
  var left = replaced.split('');
  var right = replaced.split('').reverse();
  
  for( var ii = 0; ii < left.length; ii++){
    if( left[ii] !== right[ii] ){
      return false;
    }
  }
  
  return true;
}



palindrome("1 eye for of 1 eye.");

```

**Comment by luketevl** [comment]
Hi, how are you ?

Your problem is in ".replace(/([^a-z]+)/gmi, '') " this remove the point in "1 eye for of 1 eye**.**" ! Because this "1 eye for of 1 eye**.**" is equal ""1 eye for of 1 eye" .

Try this .replace(/([^a-z.]+)/gmi, '')

Bye !

**Comment by erictleung** [comment]
@synthet1c please be careful when reading the instructions:

> You'll need to remove all **non-alphanumeric** characters (*punctuation, spaces and symbols*)...

If you get stuck with a challenge, please first ask questions to the [Help Room](https://gitter.im/FreeCodeCamp/Help). Happy coding!

**Comment by MitchelA** [comment]
@erictleung A full stop, dot, or point _is_ a punctuation character. Therefore it should be removed. "1 eye for of 1 eye." results in the palindrome "eyeforofeye" and TS is right. It should be true.

> In punctuation, the full stop (Commonwealth English) or period (American English) is a punctuation mark placed at the end of a sentence.

See wikipedia: https://en.wikipedia.org/wiki/Full_stop

**Comment by synthet1c** [comment]
@MitchelA eric is absolutely correct. the key is **non-alphanumeric**. I made an assumption that a palindrome would only contain letters from past experience, but I think this was to see if you read the question correctly, Which I didn't. the correct reg `/([^a-z0-9]+)/gmi`produces the correct answer.

**Comment by terryc129** [comment]
Here is what I found on wiki for the definition of Palindrome 
[https://en.wikipedia.org/wiki/Palindrome](url)
> Sentences and phrases
Palindromes often consist of a sentence or phrase, e.g., "Eva, can I stab bats in a cave?", "Mr. Owl ate my metal worm", "Was it a car or a cat I saw?", "A nut for a jar of tuna", "Do geese see God?", "Ma is as selfless as I am", "On a clover, if alive erupts a vast pure evil, a fire volcano", "Dammit, I'm mad!","Dog, as a devil deified, lived as a god.", "A Toyota's a Toyota", "Go hang a salami, I'm a lasagna hog", "A Santa lived as a devil at NASA", and "An igloo! Cool, Gina!".
Punctuation, capitalization, and spaces are usually ignored. Some, such as "Rats live on no evil star", "Live on time, emit no evil", and "Step on no pets", include the spaces.

If I do not understand wrong, the fullstop shall be ignored when we test a sentence. Thus  "1 eye for of 1 eye."  is a Palindromes in this sense. Anyway this is just a minor change, and not affecting the task.But I still wish we can modify the challenge for the rest of the learner. Thanks :)
